devito/passes/clusters/blocking.py

```diff
@@ -157,3 +160,71 @@ def decompose(ispace, d, block_dims):
     directions.update({bd: ispace.directions[d] for bd in block_dims})
 
     return IterationSpace(intervals, sub_iterators, directions)
+
+
+class Skewing(Queue):
+
+    """
+    Create a new IterationSpace and skew expressions
+    Skew the accesses along a SEQUENTIAL Dimension.
+    Reference:
+    https://link.springer.com/article/10.1007%2FBF01407876
+    Example:
+
+    Transform
+
+    for i = 2, n-1
+        for j = 2, m-1
+            a[i,j] = (a[a-1,j] + a[i,j-1] + a[i+1,j] + a[i,j+1]) / 4
+        end for
+    end for
+
+    to
+
+    for i = 2, n-1
+        for j = 2+i, m-1+i
+            a[i,j-i] = (a[a-1,j-i] + a[i,j-1-i] + a[i+1,j-i] + a[i,j+1-i]) / 4
+        end for
+    end for
+    """
+
+    def callback(self, clusters, prefix):
+        if not prefix:
+            return clusters
+
+        d = prefix[-1].dim
+
+        processed_clusters = []
+        for c in clusters:
+            # Explore skewing possibilities only in case prefix is parallel and
+            # is not block-incrementing
+            if PARALLEL in c.properties[d] and not d.symbolic_incr.is_Symbol:
+                mapper, intervals = {}, []
+                skew_dim = None
+                # Traverse cluster's iteration space
+                for n, i in enumerate(c.ispace):
+                    # Identify a skew_dim and its position
+                    if (SEQUENTIAL in c.properties[i.dim]
+                       and not i.dim.symbolic_incr.is_Symbol):
+                        skew_dim = i.dim
+                        intervals.append(i)
+                    # Since we are here, prefix is skewable and nested under a
+                    # SEQUENTIAL loop. Do not skew innermost loop
+                    elif (i.dim == d and not i.dim == c.ispace[-1].dim and
```

Comment:
Avoid skewing for innermost loop

Reply:
Maybe not, add a note to fix

Reply:
```
i.dim == c.ispace[-1].dim
```

As I wrote above, you probably just want to skew the TILABLE ones, rather than using "complicated" conditions such as this one